### DISCUSSION TITLE
Slack "Failed to obtain access token"

### DISCUSSION THREAD

**Discussion by elythh** [outline/outline#3501] [Self Hosting] [answered]
Hi, since a few days I can't log into a new device/browser into my selfhosted Outline using Slack. The logs are

`{"error":"Failed to obtain access token","stack":"InternalOAuthError: Failed to obtain access token\n    at Strategy.OAuth2Strategy._createOAuthError (/opt/outline/node_modules/passport-oauth2/lib/strategy.js:423:17)\n    at /opt/outline/node_modules/passport-oauth2/lib/strategy.js:177:45\n    at /opt/outline/node_modules/oauth/lib/oauth2.js:191:18\n    at ClientRequest.<anonymous> (/opt/outline/node_modules/oauth/lib/oauth2.js:162:5)\n    at ClientRequest.emit (node:events:526:28)\n    at ClientRequest.emit (node:domain:475:12)\n    at TLSSocket.socketErrorListener (node:_http_client:442:9)\n    at TLSSocket.emit (node:events:526:28)\n    at TLSSocket.emit (node:domain:475:12)\n    at emitErrorNT (node:internal/streams/destroy:157:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:122:3)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","level":"error","message":"Error during authentication"}` 

I can't find what's causing this, the account I'm already logged to work perfectly.

**Comment by elythh** [comment, marked as answer]
Well it was also DNS, see [here](https://github.com/outline/outline/discussions/3495)

**Comment by Dorlas** [comment]
I ran into this problem in September, but only recently figured out how to solve it.
So - if you have a virtual machine, and there is a white address on it, 
then there are usually no problems with DNS.
BUT, if you have a gray address on your virtual machine, 
and you forward ports 80 and 443 through nat, this is where 
the problem with dns is.
The total solution that helped me already in 3 different configurations 
is to write all our white dns names (wiki, minio, keyloak, etc) 
in /ets/hosts and associate them with a real gray address physically assigned on the virtual machine.
The result - outside the address is white, and white addresses, 
but the virtual machine itself sees the dns addresses inside itself.
